### PR TITLE
Dockerfile - bump golang to 1.19.4 and envoy proxy 1.24.1

### PR DESCRIPTION
**Pull request by david-yu**
1.24.1 release notes: https://www.envoyproxy.io/docs/envoy/v1.24.1/version_history/v1.24/v1.24.1

### CODE REVIEW

File: .changelog/64.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (thisisnotashwin) [inline .changelog/64.txt:1]: Should we mention the change to the Dockerfile in the changelog?

**Review by thisisnotashwin — APPROVED** [review]
Thanks for doing this David! 🙏

File: .changelog/64.txt (hunk 1 above)
Comment (david-yu) [reply .changelog/64.txt]: Yes! I don't know why I forgot to commit the release note message. Thanks! Updated.

**Review by thisisnotashwin — APPROVED** [review]
TYSM!

**Pull request merged**